fastify.js

```diff
@@ -662,15 +682,32 @@ function build (options) {
     }
   }
 
+  function addSchema (schema, keyRef) {
+    if (this._schemaResolver === defaultSchemaResolver) {
+      ajv.addSchema(schema, keyRef)
+    }
+    allSchemas[keyRef || schema.$id || schema.id] = schema
+    return this
+  }
+
   function setSchemaCompiler (schemaCompiler) {
     this._schemaCompiler = schemaCompiler
     return this
   }
 
+  function setSchemaResolver (schemaResolver) {
+    this._schemaResolver = schemaResolver
+    return this
+  }
+
   function setErrorHandler (func) {
     this._errorHandler = func
     return this
   }
+
+  function getSchema (keyRef) {
+    return fastify._schemaResolver(keyRef, allSchemas)
+  }
```

Comment:
Can we avoid caching the resolved function completely? I mean, this will only be executed at startup time, and it looks like a premature optimization to cache them.